webrtc/dtls/dtls_transport.nim

```diff
@@ -0,0 +1,179 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import times, deques, tables, sequtils
+import chronos, chronicles
+import ./[dtls_utils, dtls_connection], ../errors,
+       ../stun/[stun_connection, stun_transport]
+
+import mbedtls/ssl
+import mbedtls/ssl_cookie
+import mbedtls/ssl_cache
+import mbedtls/pk
+import mbedtls/md
+import mbedtls/entropy
+import mbedtls/ctr_drbg
+import mbedtls/rsa
+import mbedtls/x509
+import mbedtls/x509_crt
+import mbedtls/bignum
+import mbedtls/error
+import mbedtls/net_sockets
+import mbedtls/timing
+
+logScope:
+  topics = "webrtc dtls"
+
+# Implementation of a DTLS client and a DTLS Server by using the Mbed-TLS library.
+# Multiple things here are unintuitive partly because of the callbacks
+# used by Mbed-TLS and that those callbacks cannot be async.
+
+# -- Dtls --
+
+type
+  Dtls* = ref object of RootObj
+    connections: Table[TransportAddress, DtlsConn]
+    transport: Stun
+    laddr*: TransportAddress
```

Comment:
I don't understand. As I said in the comment, `laddr` is already available on `transport`. It is about creating a proc to return it instead of creating a new Dtls property. The way it is now it's probably even wrong, as `laddr` is public and can be overridden resulting in conflicting values.

Reply:
I change #18 to add `raddr` to it and to be more precise. I agree it's not well done, I changed it for DTLS but I have to change it for Stun and Udp aswell. I'll open a new PR soon.